### ISSUE TITLE
[control-has-associated-label]: Does not allow for custom components that would result in accessible labels

### ISSUE THREAD

**Issue by eddhurst** [jsx-eslint/eslint-plugin-jsx-a11y#981]
This rule doesn't support dynamic components that can't be statically analysed as providing a label, but would result in an accessible label at run time.

Specifically in my instance, the react-intl component `<FormattedMessage />` takes the prop `id` and then results in the HTML structure

```html
<span>Locale-aware content</span>
```

However, the `control-has-associated-label` has no way to be extended and accept an array of components that can be safely considered as providing a label.

![Screenshot 2024-04-22 at 13 42 10](https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/assets/9933592/5f3f0e83-c4a3-4594-942b-143eb0231716)

The config only allows for adding components that should be parsed by this rule, and allows for removing components that should be excluded, but doesn't allow for extending to match a given component that would then result in a pass.

**Comment by ljharb** [comment]
This is a duplicate of #863, i believe.
